pkg/scale/result.go

```diff
@@ -0,0 +1,41 @@
+package scale
+
+import (
+	"fmt"
+	"reflect"
+)
+
+type resultCache map[string]map[bool]interface{}
+type Result struct {
+	Ok  interface{}
+	Err interface{}
+}
+
+// Validate ensures Result is valid.  Only one of the Ok and Err attributes should be nil
+func (r Result) Validate() (err error) {
```

Comment:
Could this function return a boolean instead of an error? Returning a bool seems more appropriate to me since the responsibility of the function is to check some `Result` properties. The name could be `IsValid`

```
func (r Result) IsValid() bool {
  return (r.Ok == nil && r.Err != nil) || ( r.Ok != nil && r.Err == nil)
}
```

Reply:
I'm passing this error back to the caller, and using that.  If I change it to `IsValid` I would definitely use a `bool`, but then I'd have to create an error message after checking that `bool`.  I'm going to leave it as unless you feel really strongly about this.

Reply:
I don't think the responsibility to create the error is up to this function. I mean the caller has more context of what the `result` is and if the `result` is not valid it could call another function or do something else or return a more specific error.

Reply:
👍  updated